### ISSUE TITLE
Item pipe not pulling from all slots in configurable chest - 1.18.1

### ISSUE THREAD

**Issue by mnsnownutt** [AztechMC/Modern-Industrialization#268]
I am automating with some cells, using item pipes, slot locking and configurable chests.  However, I noticed some weirdness with my system, certain things were not crafting and then figured out it was because the pipes were not pulling from all slots of the configurable chest.  

I then cutoff the flow of iron plates to the configurable chest, and noticed that gears stopped crafting, because the other 2 slots that I had plates in were not being pulled from.  Screenshot explains it.  I also attached a forced crash report for all the mod info.

As soon as I moved some of the plates to the top slot, gears began crafting again until that slot was used up.

https://pastebin.com/r672cXuQ

![2022-03-08_18 30 13](https://user-images.githubusercontent.com/31514693/157349976-f3e09d12-91fb-4d5c-9dbb-d562aaebedc2.png)

**Comment by Technici4n** [comment]
Items are being moved from the chest to the chest directly, which is why the gears have a stack size of 48. (16 is inserted and extracted right away)
You can fix this by properly configuring the pipes.

**Comment by mnsnownutt** [comment]
Not sure what that means.  Does that mean you cannot use the I/O feature, but have to put in separate push and pull pipes?  

Also, why would the pipe pull gears from the chest?  There is nothing but the chest itself requiring gears.  If that is the case, why are the plates not getting pulled and put back into the chest?  Throughput limit?

**Comment by Technici4n** [comment]
You can use the I/O feature if you correctly set up the priorities for example, otherwise yeah you need separate pipes.

It tries to extract from each slot in order, and gears are the first non-empty slot whose contents can be moved somewhere else.

**Comment by mnsnownutt** [comment]
I have tried numerous combinations of priorities and it just does not work and maybe I am just doing something wrong.  Here are some screens.
First one shows assembler running out of inductors.
![2022-03-23_19 28 17](https://user-images.githubusercontent.com/31514693/159818991-c321b007-d176-4680-9deb-a77fd0032dda.png)
.
Second one shows the pipe into the assembler set with an insert priority of 2
![2022-03-23_19 28 04](https://user-images.githubusercontent.com/31514693/159819166-2fd8cb5e-b827-4dc9-b1e8-dc739bc2e3ee.png)

Third one shows configurable chest with crafting components and only the one slot of inductors is being pulled from

![2022-03-23_19 27 30](https://user-images.githubusercontent.com/31514693/159819349-f98c4569-1fd8-4192-9034-53f6ab673ff8.png)

The pipe coming out of the configurable chest was set to the default.  I then tried changing the extract priority on the pipe coming out of the configurable chest to -10, no change.  None of the inductors from the other slots got pulled.  So, what do I have set wrong?
![2022-03-23_19 27 46](https://user-images.githubusercontent.com/31514693/159819560-42b8263a-9fc0-4b56-b386-417c9110e85a.png)

**Comment by mnsnownutt** [comment]
Another example.  Polarizer running out of neodymiu
![2022-03-22_20 01 27](https://user-images.githubusercontent.com/31514693/159819857-5c4edb80-d450-4cfc-9b28-0b141c5d26c3.png)
m

Tried setting the input priority to 10 on the pipe, did not fix the issue
![2022-03-22_20 01 37](https://user-images.githubusercontent.com/31514693/159820102-c20c1a32-19f5-4222-b991-09e5055390e1.png)

Item pipe in the configurable chest is set to 0.  Tried adjusting the extract priority down, up. Nothing.  Finally gave up and had to run a separate pipe.  Also, the same thing was happening with the stainless steel ingots going into a compressor.  They would not pull and I had to run a separate pipe.

![2022-03-22_20 01 37](https://user-images.githubusercontent.com/31514693/159820245-e2d4a48f-3368-4ec4-9a65-fb4b4226cff6.png)

**Comment by Technici4n** [comment]
Make sure that the insert priority of the configurable chest is lower than that of the target.

**Comment by mnsnownutt** [comment]
It is set to 0, that is why I do not understand why items are not moving.  For example, if the config chest priority is set to 0, and the polarizer is set to 10, why would items not move?  The screenshots clearly show the priorities to the config chest are lower. 

That is why I filed a bug report in the first place, because I did have priorities set and it did not work.  I then watched a youtuber and they had the same issue and thought the priorities were bugged on item pipes as well.  Had the same issues I did with things shutting down and not feeding items.

The only way I could get it to work is to also set the extract priority to a negative number on the configurable chest.  It then started pulling items.  Maybe there is a bug with leaving it set to 0.  I have also noticed that setting them in single increments does not seem to work, but multiples of 10 seems to finally get things moving, as long as you put negative numbers in for the config chest.  When I set it to -20, all of the sudden items started flowing.

    On Thursday, March 24, 2022, 05:30:53 AM CDT, Technici4n ***@***.***> wrote:  
 
 


Make sure that the insert priority of the configurable chest is lower than that of the target.
